packages/ui/src/ui/pages/query-tracker/module/api.ts

```diff
@@ -406,13 +409,43 @@ export function getQueryResultMeta(
     return (_dispatch, getState) => {
         const state = getState();
         const {stage} = getQueryTrackerRequestOptions(state);
-        return ytApiV4Id.getQueryResults(YTApiId.readQueryResults, {
+        return ytApiV4Id.getQueryResults(YTApiId.getQueryResults, {
             parameters: {stage, query_id, result_index},
             setup: getQTApiSetup(),
         });
     };
 }
 
+export function getQueryResultMetaList(
+    query_id: string,
+    inds: number[],
+): ThunkAction<Promise<QueryResultMeta[]>, RootState, any, any> {
+    return async (_dispatch, getState) => {
+        const state = getState();
+        const {stage} = getQueryTrackerRequestOptions(state);
+        const requests: any[] = inds.map((ind) => ({
+            command: 'get_query_result',
+            parameters: {stage, query_id, result_index: ind},
+        }));
+        const resp = (await ytApiV4Id.executeBatch(YTApiId.getQueryResults, {
```

Comment:
It looks like the code doesn't handle some errors like 'Network error' (and others)

Reply:
If `executeBatch` fails, this method will return rejected promise. And I will handle error's in other places (where this function is used).

Also I decided to return full `splitedBatchResults` (not only `results` field)